unison-src/transcripts/updateio.md

```diff
@@ -0,0 +1,240 @@
+# A test for old versions of `IO`
+
+Unison code written against older versions of the built-in `IO` type should keep
+working. This test that they do.
+
+## Test current IO
+
+First let's test that the current version works:
+
+```unison
+use .base
+
+testio.foo = '(io.printLine "Hello")
+```
+
+Let's see if it works:
+
+```ucm
+.> run testio.foo
+```
+
+Great! Now let's add an old version of `IO`:
```

Comment:
Is there any reasonable, tidier way of referencing the old supported versions of `.builtin.io`?  I feel like `.builtin` should include names for whatever the `ucm` binary understands; including old versions of IO.

Option 1: make `.builtin` a branch `Branch m`, with its own history, and then reference the abilities hash-qualified?  
Option 2: add old versions of `IO` as `.builtin.io.v1`, `.builtin.io.v2`, ..., and `.builtin.io` includes all of `vN` at the top level, where `vN` is the latest?

Reply:
I think `builtin` will include (historical) names for all old versions of `IO`, even without us doing anything. If I'm writing new code, it uses the latest `IO`. If I acquire old code that uses old `IO`, that also pulls in the history which will give the old `IO` nice hash qualified names.

I don't really think old and new IO both need nice names in the current namespace by default, but users could always `alias.type` the old one if they wish.

Reply:
(recapping our discussion) Acquiring old code that uses old IO doesn't currently necessarily pull in a historical name for `IO`, though that could change in the future (#929).